src/Map.js

```diff
@@ -258,6 +264,30 @@ export class MapGL extends Evented {
         this.getMapGL().getCanvas().style.cursor = feature ? 'pointer' : ''
     }
 
+    // Remove rendered class if rendering is happening
+    onRender = () => {
+        this._removeClass(renderedClass)
+
+        if (this._renderTimeout) {
+            clearTimeout(this._renderTimeout)
+            this._renderTimeout = null
+        }
+    }
+
+    // Add rendered class if map is idle
+    onIdle = () => {
+        // Return if some layers are still loading data
+        if (this.getLayers().some(layer => layer._isLoading)) {
+            return
+        }
+
+        // Make sure the map stay rendered for at least 500ms
+        this._renderTimeout = setTimeout(() => {
+            this._addClass(renderedClass)
+            this._renderTimeout = null
+        }, 500)
```

Comment:
@turban I don't fully understand this, probably because of my ignorance reg. the fundamentals of this app. Does a single layer emit multiple render/idle/load events before its done? Or does each layer go through a clear sequential process?

If the former is the case, then I also don't see another option as what you've done here. But, if the latter is the case, then wouldn't it be possible to wait until you have received as many `idle` events as there are layers?

And if we do need a timeout, are you sure we need half a second? I'm asking because a shorter waiting time would be beneficial for the push analytics service performance.

Reply:
It's a bit complex - some layers load additional data and might send multiple render/idle/load events. Each layer is handled individually there is no clear sequence - multiple things are loading and rendered in parallell. 

We could try to reduce the timeout, but I'm a bit worried for slower machines.